unity/Assets/Scripts/IK_Robot_Arm_Controller.cs

```diff
@@ -517,17 +517,35 @@ public void DropObject()
             jointMeta.rootRelativePosition = FirstJoint.InverseTransformPoint(joint.position);
 
             //local rotation currently relative to immediate parent joint
-            joint.GetChild(0).localRotation.ToAngleAxis(out angleRot, out vectorRot);//getchild to grab the angler since that is what actually changes the geometry angle
-            jointMeta.localRotation = new Vector4(vectorRot.x, vectorRot.y, vectorRot.z, angleRot);
+            if (joint.rotation != joint.GetChild(0).rotation)
+            {
+                joint.GetChild(0).localRotation.ToAngleAxis(out angleRot, out vectorRot);//getchild to grab the angler since that is what actually changes the geometry angle
+                jointMeta.localRotation = new Vector4(vectorRot.x, vectorRot.y, vectorRot.z, angleRot);
+            }
 
+            //edge case for where angler and parent rotations are aligned, which Quaternions have trouble resolving, so we hard-code it here
+            else
+            {
+                jointMeta.localRotation = new Vector4(1, 0, 0, 0);
+            }
+            
             //world relative rotation
             joint.GetChild(0).rotation.ToAngleAxis(out angleRot, out vectorRot);//getchild to grab the angler since that is what actually changes the geometry angle
             jointMeta.rotation = new Vector4(vectorRot.x, vectorRot.y, vectorRot.z, angleRot);
 
             //rotation relative to root joint/agent
             //root forward and agent forward are always the same
-            Quaternion.Euler(FirstJoint.InverseTransformDirection(joint.GetChild(0).eulerAngles)).ToAngleAxis(out angleRot, out vectorRot);//getchild to grab the angler since that is what actually changes the geometry angle
-            jointMeta.rootRelativeRotation = new Vector4(vectorRot.x, vectorRot.y, vectorRot.z, angleRot);
+            if (FirstJoint.rotation != joint.GetChild(0).rotation)
+            {
+                Quaternion.Euler(FirstJoint.InverseTransformDirection(joint.GetChild(0).eulerAngles)).ToAngleAxis(out angleRot, out vectorRot);//getchild to grab the angler since that is what actually changes the geometry angle
+                jointMeta.rootRelativeRotation = new Vector4(vectorRot.x, vectorRot.y, vectorRot.z, angleRot);
+            }
+
+            //edge case for when angler and root rotations are aligned, which Quaternions have trouble resolving, so we hard-code it here
+            else
+            {
+                jointMeta.rootRelativeRotation = new Vector4(1, 0, 0, 0);
+            }
```

Comment:
Small style comments, can we make this:
```c#
if (FirstJoint.rotation != joint.GetChild(0).rotation) {
    // get child to grab the angler since that is what actually changes the geometry angle
    Quaternion.Euler(FirstJoint.InverseTransformDirection(joint.GetChild(0).eulerAngles)).ToAngleAxis(out angleRot, out vectorRot);
    jointMeta.rootRelativeRotation = new Vector4(vectorRot.x, vectorRot.y, vectorRot.z, angleRot);
} else {
    // edge case for when angler and root rotations are aligned, which Quaternions have trouble resolving, so we hard-code it here
    jointMeta.rootRelativeRotation = new Vector4(1, 0, 0, 0);
}
```

It scares me putting so many newlines. I feel like I might accidently add something below the `if` and above the `else`.

It's also more consistent with some newer changes in #584.


(Same thing goes for the other snippet.)

Reply:
Easy